src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -700,32 +700,86 @@ public void ExecuteCodeBlockActions<TLanguageKindEnum>(
             CancellationToken cancellationToken)
             where TLanguageKindEnum : struct
         {
-            ExecuteBlockActionsCore<CodeBlockStartAnalyzerAction<TLanguageKindEnum>, CodeBlockAnalyzerAction, SyntaxNodeAnalyzerAction<TLanguageKindEnum>, SyntaxNode, TLanguageKindEnum>(
-                codeBlockStartActions, codeBlockActions, codeBlockEndActions, analyzer,
-                declaredNode, declaredSymbol, executableCodeBlocks, (codeBlocks) => codeBlocks.SelectMany(
-                    cb =>
+            Debug.Assert(!executableCodeBlocks.IsEmpty);
+
+            // The actions we discover in 'addActions' and then execute in 'executeActions'.
+            var ephemeralActions = ArrayBuilder<SyntaxNodeAnalyzerAction<TLanguageKindEnum>>.GetInstance();
+            ExecuteBlockActionsCore(
+                startActions,
+                actions,
+                endActions,
+                analyzer,
+                declaredNode,
+                declaredSymbol,
+                operationBlocks: [],
+                semanticModel,
+                filterSpan,
+                isGeneratedCode,
```

Comment:
note: a lot of these functions pass the same 5 arguments throughout many helpers.  I have a followup PR https://github.com/dotnet/roslyn/pull/76921 that merges those into a single type to make the data passing much simpler throughout all of this.